src/main/java/core/security/CinemaUserDetailsService.java

```diff
@@ -0,0 +1,30 @@
+package core.security;
+
+import core.model.Role;
+import core.model.User;
+import core.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CinemaUserDetailsService implements UserDetailsService {
+    private UserService userService;
+    
+    public CinemaUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+    
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException("Such user wasn't found!"));
+        return org.springframework.security.core.userdetails.User
```

Comment:
```suggestion
        return org.springframework.security.core.userdetails.User
```
When you create UserBuilder, move package name from method to imports.

Reply:
Через те що два класи мають однакове ім'я приходиться прописувати повний шлях

Reply:
розумію, але в нашого класа юзер коротший шлях буде)